store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java

```diff
@@ -325,55 +320,83 @@ public Set<StoreTrait> getTraits() {
+                rtn = getDefaultGraphs(user, operation);
+            } else {
+                if (graphIds.isEmpty()) {
+                    LOGGER.info("A get graphs request was made with empty graphIds");
+                }
+                String adminAuth = operation.isUserRequestingAdminUsage() ? this.getProperties().getAdminAuth() : null;
+                rtn.addAll(graphStorage.get(user, graphIds, adminAuth));
             }
+        } else {
+            LOGGER.warn("getGraphs was requested with null Operation, this will return no graphs.");
         }
-
         return rtn;
     }
 
-    public Map<String, Object> getAllGraphsAndAuths(final User user, final String graphIdsCsv) {
-        return this.getAllGraphsAndAuths(user, graphIdsCsv, false);
+    private String getKeyForProcessedFedStoreId() {
+        return FEDERATED_STORE_PROCESSED + id;
     }
 
-    public Map<String, Object> getAllGraphsAndAuths(final User user, final String graphIdsCsv, final boolean isAdmin) {
-        return isAdmin
-                ? graphStorage.getAllGraphsAndAccess(user, getCleanStrings(graphIdsCsv), this.getProperties().getAdminAuth())
-                : graphStorage.getAllGraphsAndAccess(user, getCleanStrings(graphIdsCsv));
+    private boolean addFedStoreIdToOperation(final Operation operation) {
+        final String keyForFedStoreId = getKeyForProcessedFedStoreId();
+        boolean isFedStoreIdPreexisting = false;
+        if (nonNull(operation) && !isNullOrEmpty(keyForFedStoreId)) {
+            //KEEP THIS ORDER v
+            final boolean doesOperationHavePreexistingFedStoreId = !isValueForFedStoreIdNullOrEmpty(operation, keyForFedStoreId);
+            //Check and Add FedStoreId to payload
+            final boolean doesPayloadHavePreexistingFedStoreId = (operation instanceof FederatedOperation)
+                    && addFedStoreIdToOperation(((FederatedOperation<?, ?>) operation).getUnClonedPayload());
+
+            //Add FedStoreId to current Operation.
+            operation.addOption(keyForFedStoreId, getValueForProcessedFedStoreId());
+
+            isFedStoreIdPreexisting = doesOperationHavePreexistingFedStoreId || doesPayloadHavePreexistingFedStoreId;
+            //KEEP THIS ORDER ^
+        }
+        return isFedStoreIdPreexisting;
+    }
+
+    private static boolean isValueForFedStoreIdNullOrEmpty(final Operation operation, final String fedStoreId) {
+        final boolean isValueForFedStoreIdNullOrEmpty = isNullOrEmpty(operation.getOption(fedStoreId, null));
+        if (operation.getOptions() != null && operation.getOptions().containsKey(fedStoreId) && isValueForFedStoreIdNullOrEmpty) {
+            //There is a slight difference between value null and key not found
+            LOGGER.debug(String.format("The FederatedStoreId Key has been with null value, this means the Key has been intentionally cleared for reprocessing by this FederatedStore. Key:%s", fedStoreId));
```

Comment:
> The FederatedStoreId Key has been with null value
This doesn't make sense, should it say something like 'The FederatedStoreId Key has a null value'?

Reply:
It was "The FederatedStoreId Key been found a with null value" not sure why words got dropped and jumbled.